### PR TITLE
[SPARK-47621][PYTHON][DOCS] Refine docstring of `try_sum`, `try_avg`, `avg`, `sum`, `mean`

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR refines docstring of  `try_sum`, `try_avg`, `avg`, `sum`, `mean`  with more descriptive examples.

### Why are the changes needed?

For better API reference documentation.

### Does this PR introduce _any_ user-facing change?

Yes, it fixes user-facing documentation.

### How was this patch tested?

Manually tested. GitHub Actions should verify them.

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Review by MaxGekk — COMMENTED** [review]
How about to demonstrate different behaviour of `try_` and none-`try_`? I mean `...the result is null on overflow`, in particular

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/sql/functions/builtin.py [hunk 1]
```diff
@@ -528,15 +528,55 @@ def try_avg(col: "ColumnOrName") -> Column:
...
+    >>> import pyspark.sql.functions as sf
+    >>> df = spark.createDataFrame([(1982, None), (1990, 2), (2000, 4)], ["birth", "age"])
+    >>> df.select(sf.try_avg("age")).show()
+    +------------+
+    |try_avg(age)|
+    +------------+
+    |         3.0|
+    +------------+
+
+    Example 3: Calculating the average age with None
```
Comment (MaxGekk) [inline python/pyspark/sql/functions/builtin.py:553]: What's the difference between this and the above example?

Comment (HyukjinKwon) [inline python/pyspark/sql/functions/builtin.py:553]:
```suggestion

    Example 2: Calculating the average age with None
```

File: python/pyspark/sql/functions/builtin.py [hunk 2]
```diff
@@ -720,13 +760,54 @@ def try_sum(col: "ColumnOrName") -> Column:
...
+    >>> import pyspark.sql.functions as sf
+    >>> df = spark.createDataFrame([(1982, None), (1990, 2), (2000, 4)], ["birth", "age"])
+    >>> df.select(sf.try_sum("age")).show()
+    +------------+
+    |try_sum(age)|
+    +------------+
+    |           6|
+    +------------+
+
+    Example 4: Overflow results in NULL when ANSI mode is on
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions/builtin.py:785]:
```suggestion
    Example 3: Overflow results in NULL when ANSI mode is on
```

File: python/pyspark/sql/functions/builtin.py [hunk 3]
```diff
@@ -528,15 +528,44 @@ def try_avg(col: "ColumnOrName") -> Column:
...
+    >>> import pyspark.sql.functions as sf
+    >>> df = spark.createDataFrame([(1982, None), (1990, 2), (2000, 4)], ["birth", "age"])
+    >>> df.select(sf.try_avg("age")).show()
+    +------------+
+    |try_avg(age)|
+    +------------+
+    |         3.0|
+    +------------+
+
+    Example 4: Overflow results in NULL when ANSI mode is on
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions/builtin.py:553]:
```suggestion
    Example 3: Overflow results in NULL when ANSI mode is on
```

File: python/pyspark/sql/functions/builtin.py (hunk 1 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/functions/builtin.py]: thanks for pointing it out

**Comment by zhengruifeng** [comment]
merged to master

**Pull request closed without merging**